comp/core/secrets/type.go

```diff
@@ -11,40 +11,9 @@ type SecretVal struct {
 	ErrorMsg string `json:"error,omitempty"`
 }
 
-// ResolveCallback is the callback type used by the ResolveWithCallback method to send notifications.
-//
-// ResolveCallback needs to acknowledge the notification by returning 'true' or refuse it and request another notification
-// on the parent by returning 'false'.
-//
-// When exploring a type the walker will emit notifications for each key being updated (when using
-// 'ResolveWithCallback'). Some notifier might not be able to handle all notifications in which case it returns 'false'.
-// The walker will then retry the notification from the parent key.
-//
-// This is needed when resolving yaml containing map[string]any. In the case of the datadog configuration the map key
-// might contain the character identified as the configuration delimiter.
-//
-// Example:
-// The following agent configuration:
-//
-//	process_config:
-//	  additional_endpoints:
-//	    https://app.datadoghq.com:
-//	      - some_api_key
-//
-// Calling 'Set("process_config.additional_endpoints.https://app.datadoghq.com", []string{"some_api_key"})'
-// will create the following invalid configuration:
-//
-//	process_config:
-//	  additional_endpoints:
-//	    https://app:
-//	      datadoghq:
-//	        com:
-//	          - some_api_key
-//
-// For this reason the notifier will refuse the notification for
-// "process_config.additional_endpoints.https://app.datadoghq.com" but accept the one for the parent
-// "process_config.additional_endpoints".
-type ResolveCallback func(key []string, value any) bool
+// SecretChangeCallback is the callback type used by SubscribeToChanges to send notifications
+// This callback will be called once for each time a handle at a particular path is resolved or refreshed
+type SecretChangeCallback func(handle string, path []string, oldValue, newValue any)
```

Comment:
What about `SecretChangedCallback`?

Reply:
No, I prefer to avoid past tense here. I think of the callback as something that is "currently happening" not something that has finished already and we are responding to, regardless of the implementation details around it. If you look around the agent codebase for other callbacks, there's a similar naming convention: `APITokenRenewCallback`, `scrubCallback`, `onDeleteCallback`...

Reply:
Thanks for the clarification 👍